types/jasmine/index.d.ts

```diff
@@ -1164,6 +1165,75 @@ declare namespace jasmine {
     var MAX_PRETTY_PRINT_DEPTH: number;
 
     var version: string;
+
+    interface JasmineOptions {
+        /**
+         * The path to the project's base directory. This can be absolute or relative
+         * to the current working directory. If it isn't specified, the current working
+         * directory will be used.
+         */
+        projectBaseDir?: string;
+    }
+
+    interface JasmineConfig {
+        /**
+         * Whether to fail specs that contain no expectations.
+         * @default false
+         */
+        failSpecWithNoExpectations?: boolean;
+        /**
+         * An array of helper file paths or globs that match helper files. Each path or
+         * glob will be evaluated relative to the spec directory. Helpers are loaded before specs.
+         */
+        helpers?: string[];
+        /**
+         * Specifies how to load files with names ending in .js. Valid values are
+         * "require" and "import". "import" should be safe in all cases, and is
+         * required if your project contains ES modules with filenames ending in .js.
+         * @default "require"
+         */
+        jsLoader?: "require" | "import";
+        /**
+         * Whether to run specs in a random order.
+         */
+        random?: boolean;
+        /**
+         * An array of module names to load via require() at the start of execution.
+         */
+        requires?: string[];
+        /**
+         * The directory that spec files are contained in, relative to the project base directory.
+         */
+        spec_dir?: string;
+        /**
+         * An array of spec file paths or globs that match helper files. Each path
+         * or glob will be evaluated relative to the spec directory.
+         */
+        spec_files?: string[];
+        /**
+         * Whether to stop suite execution on the first spec failure.
+         */
+        stopOnSpecFailure?: boolean;
+        /**
+         * Whether to stop each spec on the first expectation failure.
+         */
+        stopSpecOnExpectationFailure?: boolean;
+    }
+
+    interface DefaultReporterOptions {
+        timer?: any;
+        print?: (...args: any[]) => void;
+        showColors?: boolean;
+        jasmineCorePath?: string;
+        stackFilter?: any;
+        /**
+         * A function that takes a random seed and returns the command to reproduce
+         * that seed. Use this to customize the output when using ConsoleReporter
+         * in a different command line tool.
+         */
+        randomSeedReproductionCmd?: any;
+        order?: Order;
```

Comment:
stackFilter, random... order, I cannot find in the source code
https://github.com/jasmine/jasmine-npm/blob/main/lib/jasmine.js#L143

The first 4 I could. Should you remove the last 3?

Reply:
cool, that's inline instantiated func, here is the soure:
```js
jasmine.configureDefaultReporter.toString()
'function(options) {\n' +
  '  options.timer = options.timer || new this.jasmine.Timer();\n' +
  '  options.print = options.print || function() {\n' +
  '    process.stdout.write(util.format.apply(this, arguments));\n' +
  '  };\n' +
  "  options.showColors = options.hasOwnProperty('showColors') ? options.showColors : true;\n" +
  '  options.jasmineCorePath = options.jasmineCorePath || this.jasmineCorePath;\n' +
  '\n' +
  '  this.reporter.setOptions(options);\n' +
  '  this.defaultReporterConfigured = true;\n' +
  '}'
  ```

Reply:
So yes, only 4 top options, please.